sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -608,7 +609,9 @@ class Dataset[T] private[sql](
    * duplicate items.  As such, it is analogous to `UNION ALL` in SQL.
    * @since 1.6.0
    */
-  def union(other: Dataset[T]): Dataset[T] = withPlan[T](other)(Union)
+  def union(other: Dataset[T]): Dataset[T] = withPlan[T](other){ (left, right) =>
```

Comment:
We talked about this and decided that it would be way too confusing to require users to say `unionAll` or have `union` do a distinct (because RDDs, scalding, flink etc all use union as a combine).  I'd view `unionAll` as an alias for SQL people, and if we were going to get rid of one it would probably be that.